packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -203,3 +258,49 @@ class _BottomAppBarClipper extends CustomClipper<Path> {
         || oldClipper.notchMargin != notchMargin;
   }
 }
+
+class _BottomAppBarDefaultsM2 extends BottomAppBarTheme {
+  const _BottomAppBarDefaultsM2(this.context)
+    : super(
+      elevation: 8.0,
+    );
+
+  final BuildContext context;
+
+  @override
+  Color? get color => Theme.of(context).bottomAppBarColor;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - BottomAppBar
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+// Generated version v0_101
+class _BottomAppBarDefaultsM3 extends BottomAppBarTheme {
+  const _BottomAppBarDefaultsM3(this.context)
+    : super(
+      elevation: 3.0,
+      height: 80.0,
+    );
+
+  final BuildContext context;
+
+  @override
+  Color? get color => Theme.of(context).colorScheme.surface;
+
+  @override
+  Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+
+  @override
+  NotchedShape? get shape => const AutomaticNotchedShape(RoundedRectangleBorder());
```

Comment:
Hi @darrenaustin I'm really sorry to ask this to you, but I need to be sure about this.
1. Is the BAB does have a shape property? 
2. The actual Bottom AppBarTheme.shape is used for the notch, since M3 does not support nothe for the FAB so I think that this can be removed at all, what do you think?.

Reply:
Yeah, this is a bit tricky, as we don't want to break existing M2 apps. Yes the `BottomAppBar` does have a `shape` property. What that means for M3 is the question, as you point out the M3 spec doesn't include notches. I think it is reasonable to not support the notch for M3, but we should document that the `shape` parameter is not used if `useMaterial3` is set to true. If we got with this then there is no need to provide this default above as it will be ignored in the build method for M3.

That said, I am not that familiar with the notch and shape usage in the BottomAppBar. @HansMuller do you have an opinion on this issue (i.e. can we just ignore the shape parameter for M3)?

Reply:
In the interest of compatibility and flexibility, I think we should continue to support the BottomAppBar's shape and notchMargin properties. The BottomAppBar should match the M3 spec by default when useMaterial3 is true, but there's no need to defeat the current support for the shape parameter.

Reply:
I'm not finding the solution in what @HansMuller said so I'm trying to pointout something. 

The container of the `BottomAppBar` need a shape property of `ShapeBorder` type while the actual shape is `NotchedShape` which is not subtype of `ShapeBorder`. So we can't use the actual shape property for both the container and the notch. 

If we want to support a shape for the container of `BottomAppBar` than I think we need to take one of following approaches:
1. Add a new parameter. Something like: `containerShape`. 
2. Try to make a common interface for `ShapeBorder` and `NotchedShape` if this is possible. "E.g." Making a `Shape` interface or making `NotchedShape` subtype of `ShapeBorder`. 
3. Braking change. 

I don't think there is any other way. 
Thanks and sorry to get this long 😅

Reply:
So sorry I haven't been able to respond to this more quickly.

I think we could still support the notch, but just have the default M3 shape be the `const AutomaticNotchedShape(RoundedRectangleBorder())` which you already have coming from the defaults.

Instead of having it return different widgets for M3 vs M2, we just use something like:

```dart
    final CustomClipper<Path> clipper = notchedShape != null && hasFab
      ? _BottomAppBarClipper(
          geometry: geometryListenable,
          shape: notchedShape,
          materialKey: materialKey,
          notchMargin: widget.notchMargin,
        )
      : const ShapeBorderClipper(shape: RoundedRectangleBorder());
    final double elevation = widget.elevation ?? babTheme.elevation ?? defaults.elevation!;
    final double? height = widget.height ?? babTheme.height ?? defaults.height;
    final Color color = widget.color ?? babTheme.color ?? defaults.color!;
    final Color surfaceTintColor = widget.surfaceTintColor ?? babTheme.surfaceTintColor ?? defaults.surfaceTintColor!;
    final Color effectiveColor = isMaterial3 ? color : ElevationOverlay.applyOverlay(context, color, elevation);

    return PhysicalShape(
      clipper: clipper,
      elevation: elevation,
      color: effectiveColor,
      clipBehavior: widget.clipBehavior,
      child: Material(
        key: materialKey,
        type: MaterialType.transparency,
        elevation: elevation,
        surfaceTintColor: surfaceTintColor,
        child: widget.child == null
          ? null
          : SafeArea(child: widget.child!),
      ),
    );
```

So it always returns a PhysicalShape, but all the parameters come from the defaults or are specially set up for M3 if needed. That way, by default M3 gets the right values, but we still support the shape for M2.

If this doesn't make sense or If this is more than you feel like taking on, just let us know and one of the team can take this and try to land it. I will be on vacation for a couple of weeks, so I may not be able to respond until I get back.

Reply:
Hi, sorry I also wasn't able to answer,
To me I can fix this, but the only problem is that I'm actually have no more time like before if you will fix this without discard this PR. so please do it, but if you'll start from zero than I think it's better if you give me some more time. 

Thanks and best regards

Reply:
@darrenaustin - looks like your proposed changes were added here. Seems reasonable to me, what do you think?